File: Chapter 04/C#/Section_4_4_Anti_Patterns/After_Rich_Domain_Thin_Controller/After/After.Application/OrderRequest.cs
// Chapter 04/CSharp/4.4-Rich-After/After.BusinessLogic/OrderRequest.cs
using System.Collections.Generic;

namespace After.Application
{
    /// <summary>
    /// DTO for the items within a request.
    /// </summary>
    public class OrderItemRequest
    {
        public int ItemId { get; set; }
        public int Quantity { get; set; }
    }

    /// <summary>
    /// The main Request DTO.
    /// </summary>
    public class OrderRequest
    {
        public int CustomerId { get; set; }
        public List<OrderItemRequest> Items { get; set; }
    }
}

File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/section_4_4_anti_patterns/after_rich_domain_thin_controller/application/OrderRequest.java
package com.grokkingsoftwarearchitecture.chapter04.section_4_4_anti_patterns.after_rich_domain_thin_controller.application;

import java.util.List;

/**
 * DTO (Data Transfer Object) for incoming requests.
 * ARCHITECTURE NOTE: We use a specific Request DTO rather than the 
 * Domain to define our API contract. This prevents "Over-posting" 
 * attacks where a user might try to send a fake price in the JSON.
 */
public class OrderRequest {
    private int customerId;
    private List<OrderItemRequest> items;

    public OrderRequest() {
        // Default constructor for deserialization
    }

    public int getCustomerId() { return customerId; }
    public void setCustomerId(int customerId) { this.customerId = customerId; }

    public List<OrderItemRequest> getItems() { return items; }
    public void setItems(List<OrderItemRequest> items) { this.items = items; }

    public static class OrderItemRequest {
        private int itemId;
        private int quantity;

        public OrderItemRequest() {
            // Default constructor for deserialization
        }

        public int getItemId() { return itemId; }
        public void setItemId(int itemId) { this.itemId = itemId; }

        public int getQuantity() { return quantity; }
        public void setQuantity(int quantity) { this.quantity = quantity; }
    }
}